internal/httputil/access_log.go

```diff
@@ -0,0 +1,91 @@
+package httputil
+
+import (
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/TykTechnologies/tyk/request"
+)
+
+// AccessLogRecord is a representation of a transaction log in the Gateway
+type AccessLogRecord struct {
+	APIID    string
+	APIKey   string
+	OrgID    string
+	Latency  *analytics.Latency
+	Request  *http.Request
+	Response *http.Response
+}
+
+// NewAccessLogRecord returns an AccessLogRecord object
+func NewAccessLogRecord(apiID string, apiKey string, orgId string) *AccessLogRecord {
+	return &AccessLogRecord{
+		APIID:  apiID,
+		APIKey: apiKey,
+		OrgID:  orgId,
+	}
+}
+
+// Logger renders a single access log entry
+func (a *AccessLogRecord) Logger(log *logrus.Logger) *logrus.Entry {
```

Comment:
Hi, 
can we make this `Fields` method and return `logrus.Fields` instead of `*Logrus.Entry`? So that `log *logrus.Logger` need not be passed.
the caller would look like 
```
log.WithFields(accessLog.Fields()).Info()
```